src/Rector/ClassMethod/ParamConverterAttributeToMapEntityAttributeRector.php

```diff
@@ -0,0 +1,163 @@
+ *
+ * @see \Rector\Symfony\Tests\Rector\ClassMethod\ParamConverterAttributeToMapEntityAttributeRector\ParamConverterAttributeToMapEntityAttributeRectorTest
+ */
+final class ParamConverterAttributeToMapEntityAttributeRector extends AbstractRector implements MinPhpVersionInterface
+{
+    private const PARAM_CONVERTER_CLASS = 'Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter';
+
+    private const MAP_ENTITY_CLASS = 'Symfony\Bridge\Doctrine\Attribute\MapEntity';
+
+    public function __construct(
+        private readonly PhpAttributeAnalyzer $phpAttributeAnalyzer,
+        private readonly RenamedClassesDataCollector $renamedClassesDataCollector
+    ) {
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::ATTRIBUTES;
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace ParamConverter attribute with mappings with the MapEntity attribute',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeController
+{
+    #[Route('/blog/{date}/{slug}/comments/{comment_slug}')]
+    #[ParamConverter('post', options: ['mapping' => ['date' => 'date', 'slug' => 'slug']])]
+    #[ParamConverter('comment', options: ['mapping' => ['comment_slug' => 'slug']])]
+    public function showComment(
+        Post $post,
+        Comment $comment
+    ) {
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeController
+{
+    #[Route('/blog/{date}/{slug}/comments/{comment_slug}')]
+    public function showComment(
+        #[\Symfony\Bridge\Doctrine\Attribute\MapEntity(mapping: ['date' => 'date', 'slug' => 'slug'])] Post $post,
+        #[\Symfony\Bridge\Doctrine\Attribute\MapEntity(mapping: ['comment_slug' => 'slug'])] Comment $comment
+    ) {
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->phpAttributeAnalyzer->hasPhpAttribute($node, self::PARAM_CONVERTER_CLASS)) {
+            return null;
+        }
+
+        if (! $node->isPublic()) {
+            return null;
+        }
+
+        return $this->refactorParamConverter($node);
+    }
+
+    private function refactorParamConverter(ClassMethod $classMethod): ?Node
+    {
+        foreach ($classMethod->attrGroups as $attrGroup) {
+            foreach ($attrGroup->attrs as $attr) {
+                if ($this->isName($attr, self::PARAM_CONVERTER_CLASS)) {
+                    $foundAttribute[] = $attr;
+                }
+            }
+        }
+
+        foreach ($foundAttribute as $attr) {
```

Comment:
`$foundAttribute` could in theory be undefined if you dont check
```
     if (! $this->phpAttributeAnalyzer->hasPhpAttribute($node, self::PARAM_CONVERTER_CLASS)) {
            return null;
        }
```

that is why you get phpstan error: `  - '#Implicit array creation is not allowed \- variable \$foundAttribute might not exist#'`